src/common.c

```diff
@@ -107,15 +109,24 @@ char *rc_getstr (rc_handle *rh, char const *prompt, int do_echo)
 
 		if (p < buf + GETSTR_LENGTH)
 		{
-			if (do_echo && !is_term)
-				(void)write(out, &c, 1);
+			if (do_echo && !is_term) {
+				if(write(out, &c, 1) == -1) {
+					rc_log(LOG_ERR, "rc_getstr: user string write to stdout"
+					       " failed");
+				}
+			}
 			*p++ = c;
 		}
 	}
 
 	*p = '\0';
 
-	if (!do_echo || !is_term) (void)write(out, "\r\n", 2);
+	if (!do_echo || !is_term) {
+		if(write(out, "\r\n", 2) == -1) {
```

Comment:
nit: indentation.
However here the error is also return values < 2. Shouldn't we exit when these errors are detected? That's example applications so it shouldn't be a big deal.

Reply:
Addressed. I think we need not exit and carry on with processing.